src/runtime/vdom/vdom-render.ts

```diff
@@ -666,23 +673,29 @@ const updateFallbackSlotVisibility = (elm: d.RenderNode) => {
         // we need to check all of its sibling nodes in order to see if
         // `childNode` should be hidden
         for (const siblingNode of childNodes) {
-          if (siblingNode['s-hn'] !== childNode['s-hn'] || slotName !== '') {
-            // this sibling node is from a different component OR is a named
-            // fallback slot node
-            if (siblingNode.nodeType === NODE_TYPE.ElementNode && slotName === siblingNode.getAttribute('slot')) {
-              childNode.hidden = true;
-              break;
-            }
-          } else {
-            // this is a default fallback slot node
-            // any element or text node (with content)
-            // should hide the default fallback slot node
-            if (
-              siblingNode.nodeType === NODE_TYPE.ElementNode ||
-              (siblingNode.nodeType === NODE_TYPE.TextNode && siblingNode.textContent.trim() !== '')
-            ) {
-              childNode.hidden = true;
-              break;
+          // Don't check the node against itself
+          if (siblingNode !== childNode) {
```

Comment:
All I did here was wrap the existing logic in a check so we aren't hiding slot fallback content when comparing the exact same node